nservicebus/tools/migrate-to-native-delivery.md

```diff
@@ -0,0 +1,233 @@
+- check that necessary infrastructure for native delays is setup for all delayed message destinations found
+- list all the endpoints for which the tool can detect timeouts
+- calculate the amount of timeouts to migrate per endpoint
+- validate if there are timeouts the tool is [unable to migrate](migrate-to-native-delivery.md#limitations)
+
+Once this information has been reviewed, the migration process can be started.
+Even though the tool supports migrating all endpoints connected to the persistence at once, it is highly suggested to migrate endpoint by endpoint, especially for critical endpoints. Even when selecting the `--allEndpoints` option, the tool will execute an endpoint-by-endpoint migration behind the scenes.
+
+## Cleanup
+
+The tool will not delete any timeouts or tables to prevent message loss. This section describes how to cleanup archived timeouts and remove no longer used tables once the migration have been verified.
+
+WARN: This is a destructive operation and should only be performed once its been verified that the migration has been successful.
+
+### RavenDB
+
+Delete all the documents in the `TimeoutDatas` collection that have an `OwningTimeoutManager` starting with `__migrated__`.
+
+### Sql persistence
+
+Use `SELECT * FROM TimeoutsMigration_State` to list all performed migrations. For all the successful once do the following:
+
+- Make sure that [RabbitMQ compatibility mode](/transports/rabbitmq/delayed-delivery.md#backwards-compatibility) is turned off
+- Delete the `{EndpointName}_TimeoutData` table
+- Delete the migration table named `TimeoutData_migration_{MigrationRunId}`, where `MigrationRunId` is taken from the output of the `TimeoutsMigration_State` query.
+
+## Limitations
+
+As documented in the [RabbitMQ transport](/transports/rabbitmq/delayed-delivery.md), the maximum delay value of a timeout is 8,5 years. If the migration tool encounters any timeouts that have delivery time set beyond that, it won't be possible to migrate that endpoint.
+
+If the tool presents endpoints that are not part of the system when running the `preview` command, it might be that an endpoint was renamed at some point.
+Any timeouts that were stored for that endpoint, might already be late in delivery and should be handled separate from the migration tool since the tool has no way to detect where to migrate them to.
+
+## Troubleshooting
+
+If the migration started but stopped or failed along the way, the migration tool can recover and continue where it left off. To resume an interrupted migration the tool must be run with the same arguments.
+
+To run the tool with different arguments any in-progress migration needs to be aborted using the `abort` command. Any timeouts that have been fully migrated at that point will not be restored since they already have been delivered to the native timeout infrastructure. Timeouts that were scheduled to migrate will be made available again to the legacy Timeout Manager.
```

Comment:
```suggestion
To run the tool with different arguments, any in-progress migrations first need to be aborted using the `abort` command. Any timeouts that have been fully migrated at that point will not be restored since they have already been migrated to the native timeout infrastructure. Timeouts that were scheduled to migrate will be made available again to the legacy Timeout Manager.
```